packages/block-library/src/embed/index.js

```diff
@@ -183,10 +191,32 @@ export function getEmbedEdit( title, icon ) {
 				}
 
 				if ( aspectRatioClassName ) {
-					const className = classnames( this.props.attributes.className, 'wp-has-aspect-ratio', aspectRatioClassName );
-					this.props.setAttributes( { className } );
+					return {
```

Comment:
In this case though, the object is used to remove classes as well in `toggleResponsive`, so if it was a string we'd need to convert that to an object so that `classnames` could remove the classes. Unless there's an option that tells classnames to remove the classnames contained in a string? The only way I could see to do that was to pass it an object.

Reply:
Oh right. Could `allowResponsive` be passed in as an argument to this function, so that you then wouldn't need to unset individual class names? It could become:

```
const hasAspectRatioClass = allowResponsive && !! aspectRatioClassName;
return classnames( {
  [ aspectRatioClassName ]: hasAspectRatioClass,
  'wp-has-aspect-ratio': hasAspectRatioClass,
} );
```

Reply:
ah, so a flag that says if we're removing or including, that could work! :+1: